flink-connectors/flink-connector-kafka-0.8/src/main/java/org/apache/flink/streaming/connectors/kafka/internals/ClosableBlockingQueue.java

```diff
@@ -355,13 +355,13 @@ public E getElementBlocking(long timeoutMillis) throws InterruptedException {
 			throw new IllegalArgumentException("invalid timeout");
 		}
 		
-		final long deadline = System.currentTimeMillis() + timeoutMillis;
+		final long deadline = System.nanoTime() + timeoutMillis * 1_000_000L;
```

Comment:
I think it is nice to have the arguments as either milliseconds, or as `long value, TimeUnit unit`. That is the unofficial standard in Java which most users implicitly assume.